start

```diff
@@ -33,16 +37,39 @@ done
 
 cd "${projectRoot}"
 
-./ffc-sfd-landing-page/scripts/start -d;
-./ffc-sfd-auth/scripts/start -d;
-./ffc-sfd-business/scripts/start -d;
-./ffc-sfd-data/scripts/start -d;
-./ffc-sfd-permissions/scripts/start -d;
-./ffc-sfd-permissions-api/scripts/start -d;
-./ffc-sfd-ahwp-stub/scripts/start -d;
-./ffc-sfd-messages/scripts/start -d;
-./ffc-sfd-messages-processor/scripts/start -d;
-./ffc-sfd-customer-receiver-messages/scripts/start -d;
-./ffc-sfd-customer-receiver-payments/scripts/start -d;
-./ffc-sfd-customer-receiver-preferences/scripts/start -d;
-./ffc-sfd-proxy/scripts/start -d; # Must be after all web applications
+if [ "${sfd}" = "true" ]; then
+  ./ffc-sfd-data/scripts/start -d;
+  echo "Waiting for cosmosDB to start" # Cosmos is a dependency for receiver services
+  echo "Cosmos countdown:"
+  for ((i=90; i>=1; i--)); do
+    printf "\r$i"
+    sleep 1
+  done
+  printf "Cosmos countdown complete"
+  ./ffc-sfd-landing-page/scripts/start -d;
+  ./ffc-sfd-auth/scripts/start -d;
+  ./ffc-sfd-business/scripts/start -d;
+  ./ffc-sfd-permissions/scripts/start -d;
+  ./ffc-sfd-permissions-api/scripts/start -d;
+  ./ffc-sfd-ahwp-stub/scripts/start -d;
+  ./ffc-sfd-messages/scripts/start -d;
+  ./ffc-sfd-messages-processor/scripts/start -d;
+  ./ffc-sfd-customer-receiver-messages/scripts/start -d;
+  ./ffc-sfd-customer-receiver-payments/scripts/start -d;
+  ./ffc-sfd-customer-receiver-preferences/scripts/start -d;
+  ./ffc-sfd-proxy/scripts/start -d; # Must be after all web applications
+fi
+
+if [ "${data}" = "true" ]; then
+  ./ffc-sfd-data/scripts/start -d;
```

Comment:
I think `data` should start up in both scenarios as login won't work without it

Reply:
I think `ffc-sfd-data` service does start up by default, on line 41. 

Or have I misunderstood?

Reply:
It doesn't start if receivers only is selected and as that's creating a shared cosmos instance it would fail.  Or do the receivers all create cosmos too?

Reply:
Receivers only is not intended to be an option.

By default everything, then option for data and receivers or sfd but with receivers disabled

Or that was my intention. I don't think theres any value to having receivers only as I agree it needs the data service but I think the above options are covered.

The receiver services do not currently create a cosmos instance but they should to support standalone development, will create a ticket.

Reply:
The way I see this at the moment is we've essentially got two contexts here.

Single Front Door - web based services to allow users to view and amend their farming data
Customer - data integration services to support data access and storage for services like SFD

So it's very likely that we'll have scenarios where we're only interested in one of those concerns.  So being able to choose which service you want feels helpful.  For example, if I'm updating a web page on SFD, I don't want to have to wait for all the receivers services to start up and start consuming my CPU and memory if I'm not interested in them.

Grey area is the data service as it's part of the customer context but is a dependency for SFD.

As for the receivers being able to start standalone.  Yes that is definitely something we should do as they should all have strong local development stories.  However, given we're in Alpha and only prototyping potential solutions, we don't need to dedicate time to doing that now, just capturing that we need to do it should it transition to a live service.